3 - PB/Documentazione interna/Norme di Progetto/Norme di Progetto.typ

```diff
@@ -1697,43 +1697,45 @@ Gli argomenti principali trattati nel documento sono due:
 
-== Processo di implementazione
-<processo_implementazione>
+== Processo di implementazione <processo_implementazione>
 
 _Conformant to outcomes to ISO/IEC/IEEE 12207:2017 clause 6.4.7_
 
 === Scopo
 
-Lo scopo del processo di implementazione è quello di realizzare uno specificato elemento di sistema. Questo processo trasforma requisiti, architetture e design, includendo le interfacce, in azioni che creano un elemento di sistema in accordo con le prassi della tecnologia implementativa selezionata, usando appropriate specialità o discipline tecniche.
+Lo scopo del processo di implementazione è di concretizzare un elemento specifico del sistema. Questo processo traduce i requisiti, le architetture e i design (includendo le interfacce), in azioni che danno vita ad un elemento di sistema in linea con le pratiche della tecnologia implementativa selezionata, facendo uso delle specializzazioni e discipline tecniche appropriate.
 
-Questo processo risulta in un elemento di sistema che soddisfa specificati requisiti di sistema (inclusi requisiti specifici e i derivati), architetture e design.
+Il risultato di questo processo è un elemento del sistema che soddisfa i requisiti specifici e derivati del sistema, nonché le architetture e i design definiti.
 
 === Risultati
 
 Come risultato della corretta implementazione del processo di implementazione otteniamo:
 
 - identificazione dei vincoli implementativi che influenzano i requisiti, l'architettura o il design;
 - realizzazione di un elemento di sistema;
-- l'implementazione viene tracciata.
+- l'implementazione di tale elemento viene tracciata.
 
 === Attività
 
 ==== Preparazione per l'implementazione
-Fondamentale per garantire il livello di qualità richiesto dal #pdq è definire test per l'elemento di sistema che si vuole realizzare secondo il modello di sviluppo _Test Driven Development_. Questo si concretizza nella realizzazione da parte del Progettista di una serie di unit test precedentemente allo sviluppo vero e proprio dell'elemento di sistema. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
 
-In caso di modifica eseguendo nuovamente i test si garantisce che il software sviluppato e testato in precedenza funzioni ancora come previsto. In caso contrario, si parlerebbe di regressione.
+Fondamentale per garantire il livello di qualità richiesto dal #pdq è definire test specifici per l'elemento di sistema che si vuole realizzare secondo il modello di sviluppo _Test Driven Development_. Questo si concretizza nella realizzazione da parte del Progettista di una serie di unit test precedentemente allo sviluppo vero e proprio dell'elemento di sistema. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
```

Comment:
Aggiungerei un paragrafo prima di questo dove viene detto che è importante che ciò che si vuole implementare sia prima stato definito a livello di architettura e design, citando i due capitoli sui relativi processi, spiegando anche il motivo. Quindi, ad esempio, che in questo modo si ha già la struttura di quello che si deve implementare e questo processo diventa una sorta di "traduzione" da design a codice. Poi vedi te ovviamente come/cosa scrivere, questa è la prima cosa che mi è venuta in mente

È già un po' scritta nello Scopo questa cosa, però approfondirei un pochino perché si tratta sempre di qualcosa che prepara all'implementazione

Reply:
fammi sapere cosa ne pensi